ddp-workspace/projects/ddp-dsm-ui/src/app/home/home.component.ts

```diff
@@ -1,20 +1,28 @@
-import { Component } from '@angular/core';
+import {Component, OnDestroy, OnInit} from '@angular/core';
 import { Auth } from '../services/auth.service';
+import {ActivatedRoute} from '@angular/router';
+import {takeUntil} from 'rxjs/operators';
+import {Subject} from 'rxjs';
 
 @Component({
   selector: 'app-home',
   templateUrl: './home.component.html',
 })
-export class HomeComponent {
+export class HomeComponent implements OnInit, OnDestroy {
   notAllowedToLogin = false;
+  selectedRealm: string;
+  destroy = new Subject();
 
-  constructor(public auth: Auth) {
-    auth.events.subscribe(
-      e => {
-        if (e === Auth.AUTHENTICATION_ERROR) {
-          this.notAllowedToLogin = true;
-        }
-      }
-    );
+  constructor(public auth: Auth, private route: ActivatedRoute) {}
+
+  ngOnInit(): void {
+    this.auth.events.pipe(takeUntil(this.destroy)).subscribe(e => e === Auth.AUTHENTICATION_ERROR && (this.notAllowedToLogin = true));
```

Comment:
lambda expression can be extracted as method as well something like `isNotAllowedToLogin`, in Java there is a feature called `method reference` for lambdas, if we have the same feature here we could do `.subscribe(this::isNotAllowedToLogin)` which will be much cleaner